crates/editor/src/display_map.rs

```diff
@@ -983,8 +983,23 @@ impl DisplaySnapshot {
                     break;
                 }
             }
-            let end = end.unwrap_or(max_point);
-            Some((start..end, self.fold_placeholder.clone()))
+            let mut adjusted_end = end.unwrap_or(max_point);
```

Comment:
```suggestion
            let mut end_before_line_breaks = end.unwrap_or(max_point);
```

NIT: we could remove both comments in this new code and name the variable a bit more precise.

-----------

Or, we can go a bit further and stop calculating the column in the loop with 
```rust
 self.buffer_snapshot
                        .line_len(MultiBufferRow(adjusted_end.row - 1))
```
and do that only once in the end.

So the variable name above could become `row_before_line_breaks` or similar.

Reply:
Thank you for the suggestions! I have updated the code accordingly.